File: Source/Fastore.Client/TransactionIDComparer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Alphora.Fastore.Client
{
	public class TransactionIDComparer : IComparer<TransactionID>
	{
		public static readonly TransactionIDComparer Default = new TransactionIDComparer();

		public int Compare(TransactionID x, TransactionID y)
		{
			return x.Revision.CompareTo(y.Revision);
		}
	}
}

File: Source/FastoreClient/TransactionIDComparer.cpp
#include "TransactionIDComparer.h"

//C# TO C++ CONVERTER TODO TASK: The .NET System namespace is not available from native C++:
//using namespace System;
//C# TO C++ CONVERTER TODO TASK: The .NET System namespace is not available from native C++:
//using namespace System::Collections::Generic;
//C# TO C++ CONVERTER TODO TASK: The .NET System namespace is not available from native C++:
//using namespace System::Linq;
//C# TO C++ CONVERTER TODO TASK: The .NET System namespace is not available from native C++:
//using namespace System::Text;
namespace Alphora
{
	namespace Fastore
	{
		namespace Client
		{

const boost::shared_ptr<TransactionIDComparer> TransactionIDComparer::Default = boost::make_shared<TransactionIDComparer>();

			int TransactionIDComparer::Compare(const boost::shared_ptr<TransactionID> &x, const boost::shared_ptr<TransactionID> &y)
			{
				return x->Revision.compare(y->Revision);
			}
		}
	}
}
